accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -1072,10 +1072,9 @@ If _DV_ does not include an `<interfaces>` clause then _DV_ has
 relation which was specified earlier.
 
 A compile-time error occurs if `V1` is a type name or a parameterized type
-which occurs as a superinterface in an extension type declaration _DV_, but
-`V1` does not denote an extension type, and `V1` does not denote a
-supertype of the extension type erasure of the representation type of
-_DV_.
+which occurs as a superinterface in an extension type declaration _DV_, and
+`V1` denotes a non-extension type which is not a supertype of the
+representation type of _DV_.
 
 *In particular, in order to have `implements T` where `T` is a
 non-extension type, it must be sound to consider the representation object
```

Comment:
Below, use "declared direct or indirect superinterface", to distinguish the syntactic clauses of a declaration from the semantic superinterfaces implemented by the semantics of that declaration.
(Also, "is the type `Name`" should say "denotes the declaration _DV_", to better account for type aliasand shadowing. It's a compile-time error for a different reason if a type parameter is also called `Name`, but then that would be what the implements clause denotes.)

Reply:
That's much better! Done.

About the word 'declared': The only 'undeclared' superinterfaces are `Object` which may implicitly be the superclass of a class (and _will_ implicitly be the superclass of a mixin class), and `Object` as an implicit left operand of `with` (which is then implicitly a superinterface of the mixin application). I can't see how it would create a problem for the correctness of the cyclicity rule that `Object` is included in these situations, just like the explicit superinterfaces.